pkg/client/query/interface.go

```diff
@@ -0,0 +1,16 @@
+package query
+
+// ParamsCache is an interface for a simple in-memory cache implementation for query parameters.
+type ParamsCache[T any] interface {
```

Comment:
A `Param` can be represented as a `KeyValue`.

Any reason we can't consolidate the two?

Reply:
If we consolidate the `ParamsCache` and `KeyValueCache`, we'll end up with
```
ParamsCache[P]#Get(key)
ParamsCache[P]#Set(key, value)
```

Where `key` would be a made up key that we force the developer to provide.

This would leave us with the following options:
* Empty string key `paramsCache.Set("", param)`
* Any key `paramsCache.Set("anyKeyWouldWork`, param)`
* An enforced constant `paramsCache.Set(SingleValueConst, param)`

Those options were considered but opted for a keyless params cache for clarity.

Reply:
#PUC with a summary of this distinction. I also was initially questioning the need for distinct interfaces, and like your solution. :+1: